jib-core/src/test/java/com/google/cloud/tools/jib/frontend/CredentialRetrieverFactoryTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.frontend;
+
+import com.google.cloud.tools.jib.JibLogger;
+import com.google.cloud.tools.jib.configuration.credentials.Credentials;
+import com.google.cloud.tools.jib.image.ImageReference;
+import com.google.cloud.tools.jib.registry.credentials.DockerCredentialHelper;
+import com.google.cloud.tools.jib.registry.credentials.DockerCredentialHelperFactory;
+import java.nio.file.Paths;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/** Tests for {@link CredentialRetrieverFactory}. */
+@RunWith(MockitoJUnitRunner.class)
+public class CredentialRetrieverFactoryTest {
+
+  @Mock private JibLogger mockJibLogger;
+  @Mock private DockerCredentialHelperFactory mockDockerCredentialHelperFactory;
+  @Mock private DockerCredentialHelper mockDockerCredentialHelper;
+
+  @Test
+  public void testDockerCredentialHelper() throws Exception {
+    CredentialRetrieverFactory credentialProviderFactory =
+        CredentialRetrieverFactory.forImage(
+            ImageReference.of("registry", null, null), mockJibLogger);
+    Credentials expectedCredentials = new Credentials("username", "password");
+
+    Mockito.when(
+            mockDockerCredentialHelperFactory.newDockerCredentialHelper(
+                "registry", Paths.get("docker-credential-helper")))
+        .thenReturn(mockDockerCredentialHelper);
+    Mockito.when(mockDockerCredentialHelper.retrieve()).thenReturn(expectedCredentials);
+
+    Assert.assertEquals(
+        expectedCredentials,
+        credentialProviderFactory
```

Comment:
This seems the opposite of what I'd expect: I thought you'd create a DockerCredentialHelper and then ask it to get credentials for the various image references.

Reply:
Hmm... interesting point. Honestly, I'm overall not familiar how all the credential-related classes are interfaced.